fmlloader/src/main/java/net/minecraftforge/fml/loading/UniqueModListBuilder.java

```diff
@@ -66,28 +72,34 @@ public UniqueModListData buildUniqueList()
 
         // Its theoretically possible that some mod has somehow moved an id to a secondary place, thus causing a dupe.
         // We can't handle this
-        final List<IModInfo> dupedMods = modIds.values().stream()
+        final List<String> dupedModErrors = modIds.values().stream()
                 .filter(modInfos -> modInfos.size() > 1)
-                .map(modInfos -> modInfos.get(0))
-                .toList();
-
-        if (!dupedMods.isEmpty()) {
-            final List<EarlyLoadingException.ExceptionData> duplicateModErrors = dupedMods.stream()
-                                                                                   .map(dm -> new EarlyLoadingException.ExceptionData("fml.modloading.dupedmod", dm, Objects.toString(dm)))
-                                                                                   .toList();
-            throw new EarlyLoadingException("Duplicate mods found", null,  duplicateModErrors);
+                .map(mods -> String.format("\tMod ID: '%s' from mod files: %s",
+                        mods.get(0).getModId(),
+                        mods.stream()
+                                .map(modInfo -> modInfo.getOwningFile().getFile().getFileName()).collect(joining(", "))
+                )).toList();
+
+        if (!dupedModErrors.isEmpty()) {
+            LOGGER.error(LOADING, "Found duplicate mods:\n{}", dupedModErrors);
+            throw new EarlyLoadingException("Duplicate mods found", null, dupedModErrors.stream()
+                    .map(s -> new EarlyLoadingException.ExceptionData(s))
+                    .toList());
         }
 
-        final List<ModFile> dupedLibs = versionedLibIds.values().stream()
-                .filter(modFiles -> modFiles.size() > 1)
-                .map(modFiles -> modFiles.get(0))
-                .toList();
 
-        if (!dupedLibs.isEmpty()) {
-            final List<EarlyLoadingException.ExceptionData> duplicateLibErrors = dupedLibs.stream()
-                .map(dm -> new EarlyLoadingException.ExceptionData("fml.modloading.dupedlib.versioned", dm, Objects.toString(dm)))
-                .toList();
-            throw new EarlyLoadingException("Duplicate plugins or libraries found", null,  duplicateLibErrors);
+        final List<String> dupedLibErrors = versionedLibIds.values().stream()
+                .filter(modFiles -> modFiles.size() > 1)
+                .map(mods -> String.format("\tDuped lib set: %s",
```

Comment:
This is not clear, what is this supposed to mean? It should probably be formatted the same as mods unless I'm misunderstanding how this would work.

Reply:
For mods you have ModInfos, and for libs you have ModFiles. I couldn't find an identifier for ModFiles thats equivilent to  ModIDs, so I have ended up just listing them like this, using what I think is the file name.

Reply:
Sure, that's fine, though it's technically the name of the module in this case. I was more referring to `Duped lib set` as that is not clear whatsoever, same with `Found duplicate libs`. That message needs to be more clear to those reading it along with from what files.

You can pretty much just copy what you've done above, except return the name instead of the mod id as you are dealing with mod files.